src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java

```diff
@@ -0,0 +1,86 @@
+package com.dynatrace.prototype.payloadHandler;
+
+import com.dynatrace.prototype.domainModel.KeptnCloudEvent;
+import com.dynatrace.prototype.payloadCreator.*;
+import com.slack.api.Slack;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.chat.ChatPostMessageRequest;
+import com.slack.api.methods.response.chat.ChatPostMessageResponse;
+import com.slack.api.model.block.LayoutBlock;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+@ApplicationScoped
+public class SlackHandler implements KeptnCloudEventHandler {
+    private static final String ENV_SLACK_TOKEN = "SLACK_TOKEN";
+    private static final String ENV_SLACK_CHANNEL = "SLACK_CHANNEL";
+    private static final String SLACK_NOTIFICATION_MSG = "A new Keptn Event arrived.";
+
+    private LinkedHashSet<KeptnCloudEventDecorator> decorators;
+
+    public SlackHandler() {
+        this.decorators = new LinkedHashSet<>();
+
+        decorators.add(new ProjectDecorator());
+        decorators.add(new ServiceDecorator());
+        decorators.add(new ApprovalDecorator());
+        decorators.add(new DeploymentDecorator());
+        decorators.add(new TestDecorator());
+        decorators.add(new EvaluationDecorator());
+        decorators.add(new ReleaseDecorator());
+        decorators.add(new GetActionDecorator());
+        decorators.add(new GetSLIDecorator());
+        decorators.add(new ProblemDecorator());
+        //is a default decorator needed?
+    }
+
+    @Override
+    public boolean handleEvent(KeptnCloudEvent event) {
+        boolean successful = false;
+        Slack slack = Slack.getInstance();
+        String token = System.getenv(ENV_SLACK_TOKEN);
+        String channel = System.getenv(ENV_SLACK_CHANNEL);
+
+        if (token == null) {
+            System.err.println(ENV_SLACK_TOKEN +" is null!");
+        } else if (channel == null) {
+            System.err.println(ENV_SLACK_CHANNEL +" is null!");
+        } else {
+            try {
+                List<LayoutBlock> layoutBlockList = new ArrayList<>();
+                boolean foundRightDecorator = false;
+                Iterator<KeptnCloudEventDecorator> decoratorIterator = decorators.iterator();
+
+                while (!foundRightDecorator && decoratorIterator.hasNext()) {
+                    layoutBlockList.addAll(decoratorIterator.next().getSpecificData(event));
+                    if (layoutBlockList.size() > 0) {
+                        foundRightDecorator = true;
+                    }
+                }
```

Comment:
I would change this to a `foreach` loop ([docs](https://docs.oracle.com/javase/8/docs/technotes/guides/language/foreach.html)). 
I would expect that more than one could potentially apply in the future. For example, for the evaluation event, we could have one mapper that creates the text message and another one that creates an image with the heatmap of the evaluation 🤔